### ISSUE TITLE
No support for AVX512 in torch 1.11.0+cpu python package

### ISSUE THREAD

**Issue by rafariossaa** [pytorch/pytorch#74950]
### 🐛 Describe the bug

When running the following small check in a m6i AWS instance, I got in the output that it is not using **AVX512**.
```bash
root@89483cacf38c:/app# pip install torch==v1.11.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==v1.11.0+cpu
  Using cached https://download.pytorch.org/whl/cpu/torch-1.11.0%2Bcpu-cp38-cp38-linux_x86_64.whl (169.2 MB)
Requirement already satisfied: typing-extensions in /opt/bitnami/python/lib/python3.8/site-packages (from torch==v1.11.0+cpu) (4.1.1)
Installing collected packages: torch
Successfully installed torch-1.11.0+cpu

root@89483cacf38c:/app# python -c "import torch; print(torch.__config__.show())"
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.5.2 (Git Hash a9302535553c73243c632ad3c4c80beec3d19a1e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.11.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 
 ```
but it is ok with `v1.10`:
```
root@89483cacf38c:/app# pip install torch==v1.10.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==v1.10.0+cpu
  Using cached https://download.pytorch.org/whl/cpu/torch-1.10.0%2Bcpu-cp38-cp38-linux_x86_64.whl (199.3 MB)
Requirement already satisfied: typing-extensions in /opt/bitnami/python/lib/python3.8/site-packages (from torch==v1.10.0+cpu) (4.1.1)
Installing collected packages: torch
Successfully installed torch-1.10.0+cpu


root@89483cacf38c:/app# python -c "import torch; print(torch.__config__.show())"
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, 
```

My guess is that this is related with https://github.com/pytorch/pytorch/pull/66703 . 
For a comment in the code in that PR, it seems that now it is needed gcc 9 but torch v1.11 was compiled with gcc 7.3.

### Versions

pytorch v1.11.0+cpu

cc @ezyang @gchanan @zou3519 @malfet @seemethere

**Comment by agomezmoron** [comment]
Hi team,

Do you have any extra information about this and an estimation about by when could it be handled?

Thanks!

**Comment by atalman** [comment]
As per this related PR [66703](https://github.com/pytorch/pytorch/pull/66703) . We need to compile with gcc 9 in order to support AVX512.
For this we need to decommission CUDA 10.2 first since gcc 9 is not compatible with this cuda version [ref](https://stackoverflow.com/questions/6622454/cuda-incompatible-with-my-gcc-version)

**Comment by atalman** [comment]
We have same issue with conda builds 

```
(base) [ec2-user@ip-10-0-2-199 bin]$ conda install pytorch torchvision torchaudio cpuonly -c pytorch-nightly
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/ec2-user/miniconda3

  added / updated specs:
    - cpuonly
    - pytorch
    - torchaudio
    - torchvision


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    blas-1.0                   |              mkl           6 KB
    bzip2-1.0.8                |       h7b6447c_0          78 KB
    ca-certificates-2022.4.26  |       h06a4308_0         124 KB
    conda-4.12.0               |   py39h06a4308_0        14.5 MB
    cpuonly-2.0                |                0           2 KB  pytorch-nightly
    ffmpeg-4.2.2               |       h20bf706_0        59.6 MB
    freetype-2.11.0            |       h70c0345_0         618 KB
    giflib-5.2.1               |       h7b6447c_0          78 KB
    gmp-6.2.1                  |       h2531618_2         539 KB
    gnutls-3.6.15              |       he1e5248_0         1.0 MB
    intel-openmp-2021.4.0      |    h06a4308_3561         4.2 MB
    jpeg-9e                    |       h7f8727e_0         240 KB
    lame-3.100                 |       h7b6447c_0         323 KB
    lcms2-2.12                 |       h3be6417_0         312 KB
    libidn2-2.3.2              |       h7f8727e_0          81 KB
    libopus-1.3.1              |       h7b6447c_0         491 KB
    libpng-1.6.37              |       hbc83047_0         278 KB
    libtasn1-4.16.0            |       h27cfd23_0          58 KB
    libtiff-4.2.0              |       h85742a9_0         502 KB
    libunistring-0.9.10        |       h27cfd23_0         536 KB
    libuv-1.40.0               |       h7b6447c_0         736 KB
    libvpx-1.7.0               |       h439df22_0         1.2 MB
    libwebp-1.2.2              |       h55f646e_0          80 KB
    libwebp-base-1.2.2         |       h7f8727e_0         440 KB
    lz4-c-1.9.3                |       h295c915_1         185 KB
    mkl-2021.4.0               |     h06a4308_640       142.6 MB
    mkl-service-2.4.0          |   py39h7f8727e_0          59 KB
    mkl_fft-1.3.1              |   py39hd3c417c_0         182 KB
    mkl_random-1.2.2           |   py39h51133e4_0         309 KB
    nettle-3.7.3               |       hbbd107a_1         809 KB
    numpy-1.21.5               |   py39he7a7128_2          10 KB
    numpy-base-1.21.5          |   py39hf524024_2         4.9 MB
    openh264-2.1.1             |       h4ff587b_0         711 KB
    openssl-1.1.1n             |       h7f8727e_0         2.5 MB
    pillow-9.0.1               |   py39h22f2fdc_0         669 KB
    pytorch-1.12.0.dev20220505 |      py3.9_cpu_0        65.9 MB  pytorch-nightly
    pytorch-mutex-1.0          |              cpu           3 KB  pytorch-nightly
    torchaudio-0.12.0.dev20220505|         py39_cpu         5.9 MB  pytorch-nightly
    torchvision-0.13.0.dev20220505|         py39_cpu         6.3 MB  pytorch-nightly
    typing_extensions-4.1.1    |     pyh06a4308_0          28 KB
    x264-1!157.20191217        |       h7b6447c_0         922 KB
    zstd-1.4.9                 |       haebb681_0         480 KB
    ------------------------------------------------------------
                                           Total:       318.2 MB

The following NEW packages will be INSTALLED:

  blas               pkgs/main/linux-64::blas-1.0-mkl
  bzip2              pkgs/main/linux-64::bzip2-1.0.8-h7b6447c_0
  cpuonly            pytorch-nightly/noarch::cpuonly-2.0-0
  ffmpeg             pkgs/main/linux-64::ffmpeg-4.2.2-h20bf706_0
  freetype           pkgs/main/linux-64::freetype-2.11.0-h70c0345_0
  giflib             pkgs/main/linux-64::giflib-5.2.1-h7b6447c_0
  gmp                pkgs/main/linux-64::gmp-6.2.1-h2531618_2
  gnutls             pkgs/main/linux-64::gnutls-3.6.15-he1e5248_0
  intel-openmp       pkgs/main/linux-64::intel-openmp-2021.4.0-h06a4308_3561
  jpeg               pkgs/main/linux-64::jpeg-9e-h7f8727e_0
  lame               pkgs/main/linux-64::lame-3.100-h7b6447c_0
  lcms2              pkgs/main/linux-64::lcms2-2.12-h3be6417_0
  libidn2            pkgs/main/linux-64::libidn2-2.3.2-h7f8727e_0
  libopus            pkgs/main/linux-64::libopus-1.3.1-h7b6447c_0
  libpng             pkgs/main/linux-64::libpng-1.6.37-hbc83047_0
  libtasn1           pkgs/main/linux-64::libtasn1-4.16.0-h27cfd23_0
  libtiff            pkgs/main/linux-64::libtiff-4.2.0-h85742a9_0
  libunistring       pkgs/main/linux-64::libunistring-0.9.10-h27cfd23_0
  libuv              pkgs/main/linux-64::libuv-1.40.0-h7b6447c_0
  libvpx             pkgs/main/linux-64::libvpx-1.7.0-h439df22_0
  libwebp            pkgs/main/linux-64::libwebp-1.2.2-h55f646e_0
  libwebp-base       pkgs/main/linux-64::libwebp-base-1.2.2-h7f8727e_0
  lz4-c              pkgs/main/linux-64::lz4-c-1.9.3-h295c915_1
  mkl                pkgs/main/linux-64::mkl-2021.4.0-h06a4308_640
  mkl-service        pkgs/main/linux-64::mkl-service-2.4.0-py39h7f8727e_0
  mkl_fft            pkgs/main/linux-64::mkl_fft-1.3.1-py39hd3c417c_0
  mkl_random         pkgs/main/linux-64::mkl_random-1.2.2-py39h51133e4_0
  nettle             pkgs/main/linux-64::nettle-3.7.3-hbbd107a_1
  numpy              pkgs/main/linux-64::numpy-1.21.5-py39he7a7128_2
  numpy-base         pkgs/main/linux-64::numpy-base-1.21.5-py39hf524024_2
  openh264           pkgs/main/linux-64::openh264-2.1.1-h4ff587b_0
  pillow             pkgs/main/linux-64::pillow-9.0.1-py39h22f2fdc_0
  pytorch            pytorch-nightly/linux-64::pytorch-1.12.0.dev20220505-py3.9_cpu_0
  pytorch-mutex      pytorch-nightly/noarch::pytorch-mutex-1.0-cpu
  torchaudio         pytorch-nightly/linux-64::torchaudio-0.12.0.dev20220505-py39_cpu
  torchvision        pytorch-nightly/linux-64::torchvision-0.13.0.dev20220505-py39_cpu
  typing_extensions  pkgs/main/noarch::typing_extensions-4.1.1-pyh06a4308_0
  x264               pkgs/main/linux-64::x264-1!157.20191217-h7b6447c_0
  zstd               pkgs/main/linux-64::zstd-1.4.9-haebb681_0

The following packages will be UPDATED:

  ca-certificates                     2021.10.26-h06a4308_2 --> 2022.4.26-h06a4308_0
  conda                               4.11.0-py39h06a4308_0 --> 4.12.0-py39h06a4308_0
  openssl                                 1.1.1m-h7f8727e_0 --> 1.1.1n-h7f8727e_0


Proceed ([y]/n)? y


Downloading and Extracting Packages
pytorch-1.12.0.dev20 | 65.9 MB   | #################################################################################################################################### | 100% 
intel-openmp-2021.4. | 4.2 MB    | #################################################################################################################################### | 100% 
nettle-3.7.3         | 809 KB    | #################################################################################################################################### | 100% 
torchaudio-0.12.0.de | 5.9 MB    | #################################################################################################################################### | 100% 
jpeg-9e              | 240 KB    | #################################################################################################################################### | 100% 
lame-3.100           | 323 KB    | #################################################################################################################################### | 100% 
lcms2-2.12           | 312 KB    | #################################################################################################################################### | 100% 
numpy-1.21.5         | 10 KB     | #################################################################################################################################### | 100% 
lz4-c-1.9.3          | 185 KB    | #################################################################################################################################### | 100% 
openssl-1.1.1n       | 2.5 MB    | #################################################################################################################################### | 100% 
libvpx-1.7.0         | 1.2 MB    | #################################################################################################################################### | 100% 
gmp-6.2.1            | 539 KB    | #################################################################################################################################### | 100% 
freetype-2.11.0      | 618 KB    | #################################################################################################################################### | 100% 
mkl-2021.4.0         | 142.6 MB  | #################################################################################################################################### | 100% 
bzip2-1.0.8          | 78 KB     | #################################################################################################################################### | 100% 
x264-1!157.20191217  | 922 KB    | #################################################################################################################################### | 100% 
conda-4.12.0         | 14.5 MB   | #################################################################################################################################### | 100% 
blas-1.0             | 6 KB      | #################################################################################################################################### | 100% 
libopus-1.3.1        | 491 KB    | #################################################################################################################################### | 100% 
libwebp-base-1.2.2   | 440 KB    | #################################################################################################################################### | 100% 
libidn2-2.3.2        | 81 KB     | #################################################################################################################################### | 100% 
cpuonly-2.0          | 2 KB      | #################################################################################################################################### | 100% 
ffmpeg-4.2.2         | 59.6 MB   | #################################################################################################################################### | 100% 
openh264-2.1.1       | 711 KB    | #################################################################################################################################### | 100% 
gnutls-3.6.15        | 1.0 MB    | #################################################################################################################################### | 100% 
pillow-9.0.1         | 669 KB    | #################################################################################################################################### | 100% 
libtasn1-4.16.0      | 58 KB     | #################################################################################################################################### | 100% 
libunistring-0.9.10  | 536 KB    | #################################################################################################################################### | 100% 
libwebp-1.2.2        | 80 KB     | #################################################################################################################################### | 100% 
zstd-1.4.9           | 480 KB    | #################################################################################################################################### | 100% 
mkl_fft-1.3.1        | 182 KB    | #################################################################################################################################### | 100% 
libpng-1.6.37        | 278 KB    | #################################################################################################################################### | 100% 
torchvision-0.13.0.d | 6.3 MB    | #################################################################################################################################### | 100% 
ca-certificates-2022 | 124 KB    | #################################################################################################################################### | 100% 
typing_extensions-4. | 28 KB     | #################################################################################################################################### | 100% 
numpy-base-1.21.5    | 4.9 MB    | #################################################################################################################################### | 100% 
libtiff-4.2.0        | 502 KB    | #################################################################################################################################### | 100% 
mkl_random-1.2.2     | 309 KB    | #################################################################################################################################### | 100% 
pytorch-mutex-1.0    | 3 KB      | #################################################################################################################################### | 100% 
mkl-service-2.4.0    | 59 KB     | #################################################################################################################################### | 100% 
libuv-1.40.0         | 736 KB    | #################################################################################################################################### | 100% 
giflib-5.2.1         | 78 KB     | #################################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
(base) [ec2-user@ip-10-0-2-199 bin]$ python -c "import torch; print(torch.__config__.show())"
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 
  ```

**Comment by rafariossaa** [comment]
Hi @atalman ,
  Is this going to be incorporated in a release of 1.11.0 or this will appear in 1.12 ?

**Comment by atalman** [comment]
> Hi @atalman , Is this going to be incorporated in a release of 1.11.0 or this will appear in 1.12 ?

Hi @rafariossaa This will be part of 1.12 release

**Comment by rafariossaa** [comment]
Hi,
  Sorry for the delay.
  Thanks !

**Comment by rafariossaa** [comment]
Hi, I am sorry for reopening this, but it is not fixed in the publised 1.12 version:

With `1.10` (this is OK)

```
admin@ip-172-31-6-35:~$ pip install torch==v1.10.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==v1.10.0+cpu
  Downloading https://download.pytorch.org/whl/cpu/torch-1.10.0%2Bcpu-cp39-cp39-linux_x86_64.whl (199.3 MB)
     |████████████████████████████████| 199.3 MB 49 kB/s 
Collecting typing-extensions
  Downloading typing_extensions-4.3.0-py3-none-any.whl (25 kB)
Installing collected packages: typing-extensions, torch
  WARNING: The scripts convert-caffe2-to-onnx, convert-onnx-to-caffe2 and torchrun are installed in '/home/admin/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed torch-1.10.0+cpu typing-extensions-4.3.0

admin@ip-172-31-6-35:~$ python -c "import torch; print(torch.__config__.show())"
-bash: python: command not found
admin@ip-172-31-6-35:~$ python3 -c "import torch; print(torch.__config__.show())"
/home/admin/.local/lib/python3.9/site-packages/torch/package/_directory_reader.py:17: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at  ../torch/csrc/utils/tensor_numpy.cpp:68.)
  _dtype_to_storage = {data_type(0).dtype: data_type for data_type in _storages}
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX512
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, 
```

With `1.11` (this was the originally reported one)
```
admin@ip-172-31-6-35:~$ pip install torch==v1.11.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==v1.11.0+cpu
  Downloading https://download.pytorch.org/whl/cpu/torch-1.11.0%2Bcpu-cp39-cp39-linux_x86_64.whl (169.2 MB)
     |████████████████████████████████| 169.2 MB 34 kB/s 
Requirement already satisfied: typing-extensions in ./.local/lib/python3.9/site-packages (from torch==v1.11.0+cpu) (4.3.0)
Installing collected packages: torch
  Attempting uninstall: torch
    Found existing installation: torch 1.10.0+cpu
    Uninstalling torch-1.10.0+cpu:
      Successfully uninstalled torch-1.10.0+cpu
  WARNING: The scripts convert-caffe2-to-onnx, convert-onnx-to-caffe2 and torchrun are installed in '/home/admin/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed torch-1.11.0+cpu


admin@ip-172-31-6-35:~$ python3 -c "import torch; print(torch.__config__.show())"
/home/admin/.local/lib/python3.9/site-packages/torch/_masked/__init__.py:223: UserWarning: Failed to initialize NumPy: No module named 'numpy' (Triggered internally at  ../torch/csrc/utils/tensor_numpy.cpp:68.)
  example_input = torch.tensor([[-3, -2, -1], [0, 1, 2]])
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.5.2 (Git Hash a9302535553c73243c632ad3c4c80beec3d19a1e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.11.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 
```

And now with `1.12`:

```
admin@ip-172-31-6-35:~$ pip install torch==v1.12.0+cpu -f https://download.pytorch.org/whl/torch_stable.html
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting torch==v1.12.0+cpu
  Downloading https://download.pytorch.org/whl/cpu/torch-1.12.0%2Bcpu-cp39-cp39-linux_x86_64.whl (189.0 MB)
     |████████████████████████████████| 189.0 MB 53 kB/s 
Requirement already satisfied: typing-extensions in ./.local/lib/python3.9/site-packages (from torch==v1.12.0+cpu) (4.3.0)
Installing collected packages: torch
  Attempting uninstall: torch
    Found existing installation: torch 1.11.0+cpu
    Uninstalling torch-1.11.0+cpu:
      Successfully uninstalled torch-1.11.0+cpu
  WARNING: The scripts convert-caffe2-to-onnx, convert-onnx-to-caffe2 and torchrun are installed in '/home/admin/.local/bin' which is not on PATH.
  Consider adding this directory to PATH or, if you prefer to suppress this warning, use --no-warn-script-location.
Successfully installed torch-1.12.0+cpu


admin@ip-172-31-6-35:~$ python3 -c "import torch; print(torch.__config__.show())"
PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOCUPTI -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.12.0, USE_CUDA=0, USE_CUDNN=OFF, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=OFF, USE_MPI=OFF, USE_NCCL=OFF, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 
```
